x/twap/listeners.go

```diff
@@ -26,7 +28,11 @@ type epochhook struct {
 
 func (hook *epochhook) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) {
 	if epochIdentifier == hook.k.PruneEpochIdentifier(ctx) {
-		hook.k.pruneOldTwaps(ctx)
+		// TODO: configure lastKeptTime via parameter.
+		lastKeptTime := ctx.BlockTime().Add(-48 * time.Hour)
+		if err := hook.k.pruneRecordsBeforeTime(ctx, lastKeptTime); err != nil {
+			ctx.Logger().Error("Error pruning old twaps at the epoch end", err)
+		}
```

Comment:
Can we put this pruning logic in logic.go

Reply:
I think that the log and final error handling should belong at the highest level of abstraction. However, I encapsulated the pruning time logic: [04ca37d](https://github.com/osmosis-labs/osmosis/pull/2294/commits/04ca37df6615a666f569a15c351a7283d490893c). Let me know what you think